src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,366 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Method,
+  Prop,
+  State
+} from "@stencil/core";
+import { getKey } from "../../utils/key";
+import { getElementDir, hasLabel } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-rating",
+  styleUrl: "calcite-rating.scss",
+  shadow: true
+})
+export class CalciteRating {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteRatingElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /** specify the theme of scrim, defaults to light */
+  @Prop({ reflect: true }) theme: "light" | "dark";
+
+  /** specify the scale of the component, defaults to m */
+  @Prop({ reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** the value of the rating component */
+  @Prop({ reflect: true }) value = 0;
+
+  /** is the rating component in a selectable mode */
+  @Prop({ reflect: true }) readOnly = false;
+
+  /** is the rating component in a selectable mode */
+  @Prop({ reflect: true }) disabled = false;
+
+  /** display rating value */
+  @Prop({ reflect: true }) displayValue = false;
+
+  /** optionally pass a number of previous ratings to display */
+  @Prop({ reflect: true }) count?: number;
```

Comment:
@macandcheese should `count` be watched to call `determineInitialRating()`?

Reply:
This isn't actually used to compute a rating, it's just passed to the display chip if requested... we wouldn't know the value of all of those ratings to actually calculate ourselves.

Reply:
Gotcha. looks good